File: src/main/java/me/Oxygen/modules/movement/InvMove.java
package me.Oxygen.modules.movement;


import me.Oxygen.event.Event;
import me.Oxygen.event.EventTarget;
import me.Oxygen.event.events.EventTick;
import me.Oxygen.modules.Category;
import me.Oxygen.modules.Module;
import me.Oxygen.modules.ModuleRegister;
import net.minecraft.client.gui.GuiChat;
import net.minecraft.client.settings.KeyBinding;
import org.lwjgl.input.Keyboard;

@ModuleRegister(name = "InvMove", category = Category.MOVEMENT)
public class InvMove extends Module{
	
	boolean inInventory = false;
	
	@EventTarget(events = {EventTick.class})
	   private void onEvent(Event event) {
		if(event instanceof EventTick) {
			if(!(mc.currentScreen instanceof GuiChat)) {
		         if(mc.currentScreen != null) {
		            KeyBinding[] moveKeys = new KeyBinding[]{mc.gameSettings.keyBindForward, mc.gameSettings.keyBindBack, mc.gameSettings.keyBindLeft, mc.gameSettings.keyBindRight, mc.gameSettings.keyBindJump};
		            KeyBinding[] var6 = moveKeys;
		            int var5 = moveKeys.length;

		            for(int var4 = 0; var4 < var5; ++var4) {
		               KeyBinding bind = var6[var4];
		               KeyBinding.setKeyBindState(bind.getKeyCode(), Keyboard.isKeyDown(bind.getKeyCode()));
		            }

		            if(!this.inInventory) {
		               this.inInventory = !this.inInventory;
		            }
		         } else if(this.inInventory) {
		            this.inInventory = !this.inInventory;
		         }

		      }
		}
	   }

}

File: src/main/java/cn/rainbow/oxygen/module/modules/movement/InvMove.kt
package cn.rainbow.oxygen.module.modules.movement

import cn.rainbow.oxygen.event.Event
import cn.rainbow.oxygen.event.EventTarget
import cn.rainbow.oxygen.event.events.UpdateEvent
import cn.rainbow.oxygen.module.Category
import cn.rainbow.oxygen.module.Module
import cn.rainbow.oxygen.module.setting.ModeValue
import net.minecraft.client.gui.GuiChat
import net.minecraft.client.settings.KeyBinding
import net.minecraft.item.Item
import net.minecraft.item.ItemFood
import net.minecraft.item.ItemSword
import net.minecraft.network.play.client.C09PacketHeldItemChange
import org.lwjgl.input.Keyboard

class InvMove : Module("InvMove", Category.Movement) {

    val modeValue = ModeValue("Mode", "Normal", arrayOf("Normal", "Hypixel"))

    private var isWalking = false

    @EventTarget(events = [UpdateEvent::class])
    fun onUpdate(event: Event) {
        if (event is UpdateEvent) {
            if (mc.currentScreen != null && mc.currentScreen !is GuiChat) {
                isWalking = true
                if (modeValue.isCurrentMode("Hypixel")) {
                    try {
                        var i = 0
                        for (ii in 0 until 8) {
                            val stack = mc.thePlayer.inventory.getStackInSlot(i) ?: break
                            // 空手
                            if ((stack.item !is ItemFood) && (stack.item !is ItemSword) && Item.getIdFromItem(stack.item) != 345) break // 不能为Food Sword 指南针
                            i++
                        }
                        if (i == 8 && Item.getIdFromItem(mc.thePlayer.inventory.getStackInSlot(8).item) == 345) i--
                        mc.thePlayer.sendQueue.addToSendQueue(C09PacketHeldItemChange(i))
                    } catch (e: Exception) {
                        e.printStackTrace()
                    }
                }
                val moveKeys = arrayOf(
                    mc.gameSettings.keyBindForward, mc.gameSettings.keyBindBack,
                    mc.gameSettings.keyBindLeft, mc.gameSettings.keyBindRight, mc.gameSettings.keyBindJump,
                    mc.gameSettings.keyBindSprint
                )
                for (bind in moveKeys) {
                    KeyBinding.setKeyBindState(bind.keyCode, Keyboard.isKeyDown(bind.keyCode))
                }
            } else {
                if (isWalking) {
                    if (modeValue.isCurrentMode("Hypixel")) {
                        mc.thePlayer.sendQueue.addToSendQueue(C09PacketHeldItemChange(mc.thePlayer.inventory.currentItem))
                    }
                    isWalking = false
                }
            }
        }
    }

}